inginious/frontend/pages/course_admin/student_list.py

```diff
@@ -23,20 +27,22 @@ def GET_AUTH(self, courseid):  # pylint: disable=arguments-differ
         course, __ = self.get_course_and_check_rights(courseid)
 
         if "download_audiences" in flask.request.args:
-            audiences = [{"description": audience["description"],
-                           "students": audience["students"],
-                           "tutors": audience["tutors"]} for audience in
-                          self.user_manager.get_course_audiences(course)]
-            response = Response(response=yaml.dump(audiences), content_type='text/x-yaml')
-            response.headers['Content-Disposition'] = 'attachment; filename="audiences.yaml"'
+            audiences = []
+            for audience in self.user_manager.get_course_audiences(course):
+                for student in audience["students"]:
+                    audiences.append(student+",username,student,"+audience["description"])
+                for tutor in audience["tutors"]:
+                    audiences.append(tutor+",username,tutor,"+audience["description"])
+            response = Response(response=audiences, content_type='text/csv')
```

Comment:
There is no new line between the entries here. You should better rely on the ``csv.writer`` class as it is probably done for the student table exports.

Reply:
BTW, and once again actually:

> Either the id_field can be set to the email, either the user can be asked which field they prefer.

Users often complain usernames are not reliable to identify someone locally. This feature should be useful in practice.